### PR TITLE
[1.2] Fix reporting validation for observability (#231)

### LINKED ISSUE #232: [BUG] Update observability in url validation
**Describe the bug**
Observability plugin url changed from `observability` to `observability-dashboards`, the reporting validation is still expecting observability. 

In-context menu validation will fail and user won't be able to download notebook reports within menu, although they are still able to create and download notebook reports in reporting UI as a workaround.

**To Reproduce**
1. Create a notebook in observability plugin
2. Click on "Reporting actions"
3. Click on "Download PDF" and "Download PNG", network requests will fail with `[query_url]: invalid relative url`
4. Click on "Create report definition" button still works

**Expected behavior**
All buttons should work

**Plugins**
Please list all plugins currently enabled.

**Screenshots**

**Host/Environment (please complete the following information):**

**Additional context**

### PR DESCRIPTION
**Pull request by joshuali925**
Signed-off-by: Joshua Li <joshuali925@gmail.com>

### Description
Backport #231 

### Issues Resolved
#232 

### Check List
- [ ] New functionality includes testing.
  - [x] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by davidcui1225 — APPROVED** [review]
LGTM- did we get approval to introduce this into 1.2?

**Comment by joshuali925** [comment]
> LGTM- did we get approval to introduce this into 1.2?

No, just opened it in case, but don't merge it yet

**Review by kavithacm — APPROVED** [review]

**Pull request merged**
